salt/modules/ansiblegate.py

```diff
@@ -263,7 +267,7 @@ def help(module=None, *args):
             description = doc.get('description') or ''
             del doc['description']
             ret['Description'] = description
-        ret['Available sections on module "{}"'.format(module.__name__.replace('ansible.modules.', ''))] = doc.keys()
+        ret['Available sections on module "{}"'.format(module.__name__.replace('ansible.modules.', ''))] = [i for i in doc.keys()]
```

Comment:
```python
ret['Available sections on module "{}"'.format(module.__name__.replace('ansible.modules.', ''))] = list(doc)
```
?

Reply:
I had to explicitly use list comprehension here because `list(doc)` returns an empty list here even when the `doc` dict is populated... :pensive:

Reply:
Huh?!

Reply:
It's a regular dictionary! :disappointed: 
Ok, leave it.

Reply:
Yes, it's a regular dictionary.. so I'm not really sure why this is happening. However I'll try to figure out

Reply:
@s0undt3ch look at this:

```python
(Pdb) print(doc)
{'module': 'ping', 'version_added': 'historical', 'short_description': 'Try to connect to host, verify a usable python and return C(pong) on success', 'options': {'data': {'description': ['Data to return for the C(ping) return value.', 'If this parameter is set to C(crash), the module will cause an exception.'], 'type': 'str', 'default': 'pong'}}, 'seealso': [{'module': 'net_ping'}, {'module': 'win_ping'}], 'author': ['Ansible Core Team', 'Michael DeHaan']}
(Pdb) type(doc)
<class 'dict'>
(Pdb) print(list(doc))
[]
(Pdb) [i for i in doc.keys()]
['module', 'version_added', 'short_description', 'options', 'seealso', 'author']
(Pdb) print(doc.keys())
dict_keys(['module', 'version_added', 'short_description', 'options', 'seealso', 'author'])
(Pdb) print(list(doc.keys()))
[]
```

Reply:
This reminds me to WAT! :smile: 

https://www.destroyallsoftware.com/talks/wat

Reply:
This is so weird.

Reply:
The reason is in `salt.modules.ansiblegate.list` function. I'll make this change for you.

Reply:
Ah! Thanks for the clarification!